app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt

```diff
@@ -512,25 +380,6 @@ class NewMessageFragment : Fragment() {
         }
     }
 
-    private fun updateBannerVisibility() = with(binding) {
-        var externalRecipientEmail: String? = null
-        var externalRecipientQuantity = 0
-
-        listOf(toField, ccField, bccField).forEach { field ->
-            val (singleEmail, quantityForThisField) = field.findAlreadyExistingExternalRecipientsInFields()
-            externalRecipientQuantity += quantityForThisField
-
-            if (externalRecipientQuantity > 1) {
-                newMessageViewModel.externalRecipientCount.value = null to 2
-                return
-            }
-
-            if (quantityForThisField == 1) externalRecipientEmail = singleEmail
-        }
-
-        newMessageViewModel.externalRecipientCount.value = externalRecipientEmail to externalRecipientQuantity
-    }
-
```

Comment:
Line 400, you could do something like this:

```kotlin
val selectedSignature = with(signatures) {
            first {it.id == newMessageViewModel.selectedSignatureId || it.isDefault }
        }
```

to avoid the `!!` operator and to also avoid going twice through the list.

Reply:
Unfortunately, we can't simplify it this way because we first need to make sure there's no selected signature before falling back on the default one. 

This could nonetheless be simplified into a single loop but here the order of magnitude of the size of the list is of about ~10 values which is probably why we've let it go through. Do we want to optimize it further? Maybe? I don't know if it's worth the effort and the lisibility